File: AtlasWeb/Core/Modules/Tibbi_Surec/Morg_Modulu/MorgueExDischargeFormViewModel.ts
//$71912B36
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { Morgue } from "NebulaClient/Model/AtlasClientModel";
import { MorgueDeathType } from "NebulaClient/Model/AtlasClientModel";
import { DeathReasonDiagnosis } from "NebulaClient/Model/AtlasClientModel";
import { SKRSOlumSekli } from "NebulaClient/Model/AtlasClientModel";
import { SKRSICD } from "NebulaClient/Model/AtlasClientModel";
import { SKRSOlumNedeniTuru } from "NebulaClient/Model/AtlasClientModel";
import { SKRSYaralanmaninYeri } from "NebulaClient/Model/AtlasClientModel";
import { SKRSOlumYeri } from "NebulaClient/Model/AtlasClientModel";
import { ResSection } from "NebulaClient/Model/AtlasClientModel";
import { ResUser } from "NebulaClient/Model/AtlasClientModel";
import { MernisDeathReasonDefinition } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class MorgueExDischargeFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.MorgueExDischargeFormViewModel, Core';
    @Type(() => Morgue)
    public _Morgue: Morgue = new Morgue();
    @Type(() => MorgueDeathType)
    public MorgueDeathTypeGridList: Array<MorgueDeathType> = new Array<MorgueDeathType>();
    @Type(() => DeathReasonDiagnosis)
    public DeathReasonDiagnosisGridList: Array<DeathReasonDiagnosis> = new Array<DeathReasonDiagnosis>();
    @Type(() => SKRSOlumSekli)
    public SKRSOlumSeklis: Array<SKRSOlumSekli> = new Array<SKRSOlumSekli>();
    @Type(() => Morgue)
    public Morgues: Array<Morgue> = new Array<Morgue>();
    @Type(() => SKRSICD)
    public SKRSICDs: Array<SKRSICD> = new Array<SKRSICD>();
    @Type(() => SKRSOlumNedeniTuru)
    public SKRSOlumNedeniTurus: Array<SKRSOlumNedeniTuru> = new Array<SKRSOlumNedeniTuru>();
    @Type(() => SKRSYaralanmaninYeri)
    public SKRSYaralanmaninYeris: Array<SKRSYaralanmaninYeri> = new Array<SKRSYaralanmaninYeri>();
    @Type(() => SKRSOlumYeri)
    public SKRSOlumYeris: Array<SKRSOlumYeri> = new Array<SKRSOlumYeri>();
    @Type(() => ResSection)
    public ResSections: Array<ResSection> = new Array<ResSection>();
    @Type(() => ResUser)
    public ResUsers: Array<ResUser> = new Array<ResUser>();
    @Type(() => MernisDeathReasonDefinition)
    public MernisDeathReasonDefinitions: Array<MernisDeathReasonDefinition> = new Array<MernisDeathReasonDefinition>();
    public _selectedDeathType: Object;
    public _selectedDeathReason: Object;
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Morg_Modulu/MorgueExDischargeFormViewModel.cs
//$173CAF5C
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Microsoft.AspNetCore.Mvc;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class MorgueServiceController
    {
        partial void PreScript_MorgueExDischargeForm(MorgueExDischargeFormViewModel viewModel, Morgue morgue, TTObjectContext objectContext)
        {
            if (Common.CurrentResource.TakesPerformanceScore == true)
            {
                viewModel._Morgue.DoctorFixed = Common.CurrentResource;
            }
            if (morgue.CurrentStateDefID == Morgue.States.Request)
            {
                if (morgue.DiedClinic == null)
                {
                    Guid? selectedEpisodeActionObjectID = viewModel.GetSelectedEpisodeActionID();
                    var episodeAction = objectContext.GetObject<EpisodeAction>((Guid)selectedEpisodeActionObjectID);
                    if (episodeAction != null)
                    {
                        morgue.DiedClinic = episodeAction.MasterResource;
                        viewModel._Morgue.DiedClinic = episodeAction.MasterResource;
                    }
                }
            }

            ContextToViewModel(viewModel, objectContext);
        }

        partial void PostScript_MorgueExDischargeForm(MorgueExDischargeFormViewModel viewModel, Morgue morgue, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            morgue.Episode.Patient.DeathReportNo = viewModel._Morgue.DeathReportNo;
            morgue.Episode.Patient.ExDate = viewModel._Morgue.DateTimeOfDeath;
        }

        [HttpGet]
        public SKRSOlumSekli[] GetSKRSOlumSekli()
        {
            SKRSOlumSekli[] result;
            using (var objectContext = new TTObjectContext(true))
            {
                result = objectContext.QueryObjects<SKRSOlumSekli>().ToArray();
                objectContext.FullPartialllyLoadedObjects();
            }

            return result;
        }

        [HttpGet]
        public SKRSICD[] GetSKRSICDO()
        {
            SKRSICD[] result;
            using (var objectContext = new TTObjectContext(true))
            {
                result = objectContext.QueryObjects<SKRSICD>().ToArray();
                objectContext.FullPartialllyLoadedObjects();
            }

            return result;
        }

        [HttpGet]
        public SKRSOlumNedeniTuru[] GetSKRSOlumNedeni()
        {
            SKRSOlumNedeniTuru[] result;
            using (var objectContext = new TTObjectContext(true))
            {
                result = objectContext.QueryObjects<SKRSOlumNedeniTuru>().ToArray();
                objectContext.FullPartialllyLoadedObjects();
            }

            return result;
        }

    }
}

namespace Core.Models
{
    public partial class MorgueExDischargeFormViewModel
    {
        public object _selectedDeathType
        {
            get;
            set;
        }
        public object _selectedDeathReason
        {
            get;
            set;
        }

        public void AddMorgueViewModelToContext(TTObjectContext objectContext, EpisodeAction masterAction)
        {

            var morgueImported = (Morgue)objectContext.AddObject(this._Morgue);
            morgueImported.FromResource = masterAction.MasterResource;
            morgueImported.MasterAction = masterAction;// Linkactions için set edildi
            morgueImported.Episode = masterAction.Episode;
            morgueImported.CurrentStateDefID = Morgue.States.Request;
            morgueImported.SenderDoctor = masterAction.ProcedureDoctor;
            if (masterAction is TreatmentDischarge)
                morgueImported.ActionDate = ((TreatmentDischarge)masterAction).DischargeDate;
            else
                morgueImported.ActionDate = masterAction.ActionDate;
            morgueImported.RequestDate = DateTime.Now;



            if (this.MorgueDeathTypeGridList != null)
            {
                foreach (var item in this.MorgueDeathTypeGridList)
                {
                    var deathTypeImported = (MorgueDeathType)objectContext.AddObject(item);
                    deathTypeImported.Morgue = morgueImported;
                }
            }

            if (this.DeathReasonDiagnosisGridList != null)
            {
                foreach (var item in this.DeathReasonDiagnosisGridList)
                {
                    var deathReasonImported = (DeathReasonDiagnosis)objectContext.AddObject(item);
                    deathReasonImported.Morgue = morgueImported;
                }
            }


        }
    }
}
